process/block/preprocess/transactions.go

```diff
@@ -619,33 +618,40 @@ func (txs *transactions) CreateAndProcessMiniBlock(
 				log.Debug("revert to snapshot", "error", err.Error())
 			}
 
+			txs.gasHandler.RemoveGasConsumed([][]byte{orderedTxHashes[index]})
+			txs.gasHandler.RemoveGasRefunded([][]byte{orderedTxHashes[index]})
+
 			continue
 		}
 
 		err = txs.computeGasConsumed(
 			miniBlock.SenderShardID,
 			miniBlock.ReceiverShardID,
 			orderedTxs[index],
+			orderedTxHashes[index],
 			&gasConsumedByMiniBlockInSenderShard,
 			&gasConsumedByMiniBlockInReceiverShard)
 
 		if err != nil {
 			log.Debug(fmt.Sprintf("max gas limit is reached: %d per mini block in sender shard, %d per mini block in receiver shard, %d per block in self shard: added %d txs from %d txs\n",
 				gasConsumedByMiniBlockInSenderShard,
 				gasConsumedByMiniBlockInReceiverShard,
-				txs.gasHandler.GasConsumed(),
+				txs.gasHandler.TotalGasConsumed(),
 				len(miniBlock.TxHashes),
 				len(orderedTxs)))
 
-			txs.txsForCurrBlock.mutTxsForBlock.Lock()
-			delete(txs.txsForCurrBlock.txHashAndInfo, string(orderedTxHashes[index]))
-			txs.txsForCurrBlock.mutTxsForBlock.Unlock()
-
 			err = txs.accounts.RevertToSnapshot(snapshot)
 			if err != nil {
 				log.Error(err.Error())
 			}
 
+			txs.txsForCurrBlock.mutTxsForBlock.Lock()
+			delete(txs.txsForCurrBlock.txHashAndInfo, string(orderedTxHashes[index]))
```

Comment:
why are you deleting here the transactions? these are cleaned up on CreateBlockStarted
if you delete here you are affecting the indexing of transactions in elastic search

Reply:
This is done only in case of error when also others things would be reverted to a snapshot